### PR TITLE
[RELEASE] UnityModules 4.7.1 - ScreenTop Compatibility

### PR DESCRIPTION
**Pull request by zalo**
This PR is a combination of #1114 and Marin's #1135 to add Screen-Top service compatibility to the 4.6.0 release of UnityModules.

#1114 was included to add Advanced Dropdowns to the Inspector GUI, so partners can easily switch between Desktop, Screentop, and HMD tracking modes.

This PR also includes the advanced Server Namespace option from #1114, for allowing two (internal) 4.x services to feed data into Unity simultaneously.

### CODE REVIEW

File: Assets/Plugins/LeapMotion/Core/Scripts/LeapServiceProvider.cs [hunk 1]
```diff
@@ -461,7 +478,19 @@ public enum PhysicsExtrapolationMode {
...
+        _leapController.ClearPolicy(Controller.PolicyFlag.POLICY_OPTIMIZE_SCREENTOP);
+        _leapController.ClearPolicy(Controller.PolicyFlag.POLICY_OPTIMIZE_HMD);
+      } else if (_trackingOptimization == TrackingOptimizationMode.ScreenTop) {
+        _leapController.ClearPolicy(Controller.PolicyFlag.POLICY_DEFAULT);
+        _leapController.ClearPolicy(Controller.PolicyFlag.POLICY_OPTIMIZE_HMD);
+        _leapController.SetPolicy  (Controller.PolicyFlag.POLICY_OPTIMIZE_SCREENTOP);
+      } else if (_trackingOptimization == TrackingOptimizationMode.HMD) {
+        _leapController.ClearPolicy(Controller.PolicyFlag.POLICY_DEFAULT);
+        _leapController.ClearPolicy(Controller.PolicyFlag.POLICY_OPTIMIZE_SCREENTOP);
+        _leapController.SetPolicy  (Controller.PolicyFlag.POLICY_OPTIMIZE_HMD);
```
Comment (MattGrayUL) [inline Assets/Plugins/LeapMotion/Core/Scripts/LeapServiceProvider.cs:492]: Controller.PolicyFlag doesn't contain a definition for Controller.PolicyFlag.POLICY_OPTIMIZE_SCREENTOP. I assume it just needs to be added as a new flag at 1 << 8?

File: Assets/Plugins/LeapMotion/Internal/VRVisualizer/ScenesResources/Scripts/VisualizerManager.cs [hunk 2]
```diff
@@ -93,24 +101,39 @@ void Update()
...
-
-      // In Desktop Mode
-      if (m_PCVisualizer.activeInHierarchy)
+      if (m_controller.IsPolicySet(Controller.PolicyFlag.POLICY_OPTIMIZE_SCREENTOP))
       {
-        if (m_controller.IsPolicySet(Controller.PolicyFlag.POLICY_OPTIMIZE_HMD))
+        m_trackingText.text = String.Format(
+          "Tracking Mode: Screen-Top (Press '{0}' to switch to desktop mode)",
+          keyToSwitchViewMode);
+        if (Input.GetKeyDown(keyToSwitchViewMode))
```
Comment (MattGrayUL) [inline Assets/Plugins/LeapMotion/Internal/VRVisualizer/ScenesResources/Scripts/VisualizerManager.cs:109]: 'keyToSwitchViewMode' is currently named 'keyToToggleHMD' in this commit so these lines throw errors. I expect renaming keyToToggleHMD to keyToSwitchViewMode makes most sense?

**Review by MattGrayUL — COMMENTED** [review]
Hi John, thought i'd submit these as a review but let me know if you'd prefer I just commit these types of changes in future?

**Comment by zalo** [comment]
> Hi John, thought i'd submit these as a review but let me know if you'd prefer I just commit these types of changes in future?

Oops; sorry, just before your review I did a merge of `develop` into this PR to resolve Merge Conflicts.  What I didn't realize were that some of those conflicts were from reverting commits in this PR.  By reverting that merge as yet _another_ commit, I believe I've restored balance to this PR (and it should work without the obvious bugs that were present before).

**Pull request merged**